spec/services/cdl/eligible_item_service_spec.rb

```diff
@@ -0,0 +1,159 @@
+                       due_date: "null",
+                       status: [
+                         "Not Charged"
+                       ]
+                     }
+                   ],
+                   sortable_call_number: "NA.0203.G5.1967"
+                 }
+               ],
+               st: [
+                 {
+                   holding_id: 1_581_047,
+                   call_number: "NA200 .G38 1967",
+                   items: [
+                     {
+                       id: 1_666_779,
+                       on_cdl: "Y",
+                       on_reserve: "N",
+                       copy_number: 1,
+                       item_sequence_number: 1,
+                       temp_location: "null",
+                       perm_location: "st",
+                       circ_group_id: 14,
+                       pickup_location_code: "stcirc",
+                       pickup_location_id: 345,
+                       enum: "null",
+                       chron: "null",
+                       barcode: "32101015237520",
+                       item_type: "Gen",
+                       due_date: "null",
+                       status: [
+                         "Not Charged"
+                       ]
+                     }
+                   ]
+                 }
+               ],
+               uesrf: [
+                 {
+                   holding_id: 1_581_050,
+                   call_number: "NA203 .G5 1967",
+                   items: [
+                     {
+                       id: 1_666_782,
+                       on_cdl: "Y",
+                       on_reserve: "N",
+                       copy_number: 2,
+                       item_sequence_number: 1,
+                       temp_location: "null",
+                       perm_location: "uesrf",
+                       circ_group_id: 5,
+                       pickup_location_code: "uescirc",
+                       pickup_location_id: 356,
+                       enum: "null",
+                       chron: "null",
+                       barcode: "32101019151941",
+                       item_type: "NoCirc",
+                       due_date: "null",
+                       status: [
+                         "Not Charged"
+                       ]
+                     }
+                   ]
+                 }
+               ]
+             }.to_json,
+                 headers: { "Content-Type" => "application/json" })
+  end
+
+  describe ".item_ids" do
+    context "on_cdl is null" do
+      before do
+        stub_request(:get, "https://bibdata.princeton.edu/#{bib_id}/items")
+          .to_return(status: 200,
+                     body: { uesrf: [
+                       {
+                         holding_id: 1_581_050,
+                         call_number: "NA203 .G5 1967",
+                         items: [
+                           {
+                             id: 1_666_782,
+                             on_cdl: "null",
```

Comment:
Should we test that `on_cdl` is just missing? Or is that essentially the same?

Reply:
I wouldn't expect the item not to have an `on_cdl`. My assumption is that it will be in the `Item` table in voyager. But it doesn't hurt to test this as well.